src/overlays/actors/ovl_En_Raf/z_en_raf.h

```diff
@@ -2,16 +2,64 @@
 #define Z_EN_RAF_H
 
 #include "global.h"
+#include "objects/object_raf/object_raf.h"
+
+#define EN_RAF_GET_TYPE(thisx) (((thisx)->params >> 12) & 0xF)
+#define EN_RAF_GET_REVIVE_TIMER(thisx) (((thisx)->params >> 7) & 0x1F)
+#define EN_RAF_GET_SWITCH_FLAG(thisx) ((thisx)->params & 0x7F)
+
+typedef enum {
+    /* 0 */ EN_RAF_TYPE_NORMAL,
+    /* 1 */ EN_RAF_TYPE_ALEADY_DEAD,          // Spawns without trap, so it can't eat bombs/player
+    /* 2 */ EN_RAF_TYPE_NO_WATER_INTERACTIONS // Won't produce ripples, and particles won't produce splashes
+} EnRafType;
 
 struct EnRaf;
 
 typedef void (*EnRafActionFunc)(struct EnRaf*, GlobalContext*);
 
+typedef struct {
+    /* 0x00 */ u8 isVisible;
+    /* 0x04 */ Vec3f position;
+    /* 0x10 */ Vec3f velocity;
+    /* 0x1C */ Vec3f acceleration;
+    /* 0x28 */ Vec3s rotation;
+    /* 0x30 */ f32 scale;
+    /* 0x34 */ s16 timer;
+} EnRafParticle; // size = 0x38
+
 typedef struct EnRaf {
-    /* 0x0000 */ Actor actor;
-    /* 0x0144 */ char unk_144[0xEC];
-    /* 0x0230 */ EnRafActionFunc actionFunc;
-    /* 0x0234 */ char unk_234[0x8B0];
+    /* 0x000 */ DynaPolyActor dyna;
+    /* 0x15C */ SkelAnime skelAnime;
+    /* 0x1A0 */ Vec3s jointTable[CARNIVOROUS_LILY_PAD_LIMB_MAX];
+    /* 0x1E8 */ Vec3s morphTable[CARNIVOROUS_LILY_PAD_LIMB_MAX];
+    /* 0x230 */ EnRafActionFunc actionFunc;
+    /* 0x234 */ Vec3f limbScale[CARNIVOROUS_LILY_PAD_LIMB_MAX];
+    /* 0x2C4 */ Vec3f targetLimbScale[CARNIVOROUS_LILY_PAD_LIMB_MAX];
+    /* 0x354 */ Vec3s chewLimbRot[CARNIVOROUS_LILY_PAD_LIMB_MAX];
+    /* 0x39C */ s16 grabTarget;
+    /* 0x39E */ u8 isCurrentlyInRidingMovingState;
+    /* 0x3A0 */ f32 endFrame;
+    /* 0x3A4 */ f32 chewScale;
+    /* 0x3A8 */ f32 heightDiffFromBob;
```

Comment:
I think just call this "bobOffset` or something... otherwise given how it's next to `heightDiffFromPlayer`, the inevitable question is "who's Bob?".